src/realmd/AuthSocket.cpp

```diff
@@ -1036,6 +1058,78 @@ bool AuthSocket::_HandleXferAccept()
     return true;
 }
 
+// Verify PIN entry data
+bool AuthSocket::VerifyPinData(uint32 pin, const PINData& clientData)
```

Comment:
Not saying this code is useless by any means. This is basically a FIXED PIN function. Essentially Password2. All other cores use TOTP algorithm for verification.

https://github.com/vmangos/core/blob/development/src/realmd/AuthSocket.cpp#L663

look here, they have the option to use either. Our default is totp, so you should use that as a default for the pin check. IF you wish to add a db flag to an account to also enable fixed pin, im perfectly happy with that (note column flags is used by mangosd, so its not safe for realmd to also use it), but default should be totp when a token is used with an account

Reply:
Classic/Vanilla client does not have TOTP support, only PIN. So with this I'm faking TOTP support by entering it in the PIN pad

Reply:
Client doesnt do TOTP. Its about the code you input. All other cores input a google authenticator code (also known as totp) from https://datatracker.ietf.org/doc/html/rfc6238 instead of a fixed password pin. Vmangos does exactly that if you read the link i posted with the vanilla pin.

Reply:
Yeah, this is what I'm also doing. I just didn't add an option for fixed PIN because that's pointless

Reply:
Guess i missed that